pypeit/calibrations.py

```diff
@@ -344,236 +385,241 @@ def get_align(self):
+                                                     calib_dir=self.calib_dir, setup=setup,
+                                                     calib_id=calib_id)
+        # Save the result
+        self.msbias.to_file()
+        # Return it
         return self.msbias
 
     def get_dark(self):
         """
-        Load or generate the dark image
+        Load or generate the dark calibration frame.
 
         Returns:
-            :class:`~pypeit.images.buildimage.DarkImage`: The combined dark
-            image.
+            :class:`~pypeit.images.buildimage.DarkImage`: The processed
+            calibration image.
         """
         # Check internals
         self._chk_set(['det', 'calib_ID', 'par'])
 
-        # Prep
-        dark_files, self.master_key_dict['dark'] = self._prep_calibrations('dark')
-        # Construct the name, in case we need it
-        masterframe_name = masterframe.construct_file_name(buildimage.DarkImage,
-                                                           self.master_key_dict['dark'],
-                                                           master_dir=self.master_dir)
-
-        # Try to load?
-        if os.path.isfile(masterframe_name) and self.reuse_masters:
-            self.msdark = buildimage.DarkImage.from_file(masterframe_name)
-        elif len(dark_files) == 0:
-            self.msdark = None
-        else:
-            # TODO: If a bias has been constructed and it will be subtracted
-            # from the science images, it should also be subtracted from this
-            # image.  If it isn't, subtracting the dark will effectively lead to
-            # subtracting the bias twice.
-            # TODO: The order is such that the bpm doesn't exist yet.  But
-            # calling buildimage_fromlist will create the bpm if it isn't
-            # passed.  So calling get_dark then get_bpm unnecessarily creates
-            # the bpm twice.  Is there any reason why creation of the bpm should
-            # come after the dark, or can we change the order?
-            # Build and save it
-            self.msdark = buildimage.buildimage_fromlist(self.spectrograph, self.det,
-                                                         self.par['darkframe'], dark_files,
-                                                         bias=self.msbias)
-            self.msdark.to_master_file(masterframe_name)
+        # Find the calibrations
+        frame = {'type': 'dark', 'class': buildimage.DarkImage}
+        raw_files, cal_file, calib_key, setup, calib_id, detname \
+                = self.find_calibrations(frame['type'], frame['class'])
 
-        # Return
+        if len(raw_files) == 0 and cal_file is None:
+            msgs.warn(f'No raw {frame["type"]} frames found and unable to identify a relevant '
+                      'processed calibration frame.  Continuing...')
+            self.msdark = None
+            return self.msdark
```

Comment:
This is a good point.  FWIW, this code maintains the previous functionality (I think).  All of the `get_*` functions in the `Calibrations` class are called, regardless of whether or not those calibrations are needed.  The easiest option would be to remove this warning because we issue them inconsistently and they're not really that useful (as you point out).  As it is, the code will throw an error when a specific calibration frame is requested by these `use_*` parameters but none exists, which is the more critical message.

Reply:
Yeah, the functionality is fine... just trying to get rid of spurious red messages in the output stream.  Thanks for all your work on this refactor!